fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraLdp.java

```diff
@@ -663,6 +663,13 @@ private void checkAclUriExists(final URI resourceAcl) {
 
     private void addResourceAcl(final URI resourceAcl) {
         if (resourceAcl != null) {
+            final String aclHost = resourceAcl.getHost();
```

Comment:
It seems like the other checking of correctness for the ACL link is taking place in: 
https://github.com/fcrepo4/fcrepo4/blob/master/fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraLdp.java#L643

Maybe changing the name of that method and performing the check there would be more consistent.
Additionally, doing this check sooner in `PUT`/`POST` provides quicker/more efficient response.